test/integration/controllermanager/shoot/maintenance/maintenance_test.go

```diff
@@ -315,30 +325,66 @@ var _ = Describe("Shoot Maintenance controller tests", func() {
-		})
+				Expect(waitForExpectedMachineImageMaintenance(ctx, logger, testClient, shoot, testMachineImage, false, time.Now().Add(time.Second*10))).To(Succeed())
+			})
 
-		It("Shoot machine image must be updated in maintenance time: AutoUpdate.MachineImageVersion == true && expirationDate does not apply", func() {
-			// set test specific shoot settings
-			patch := client.MergeFrom(shoot.DeepCopy())
-			shoot.Spec.Maintenance.AutoUpdate.MachineImageVersion = true
-			Expect(testClient.Patch(ctx, shoot, patch)).To(Succeed())
+			It("Shoot machine image must be updated in maintenance time: AutoUpdate.MachineImageVersion == true && expirationDate does not apply", func() {
+				// set test specific shoot settings
+				patch := client.MergeFrom(shoot.DeepCopy())
+				shoot.Spec.Maintenance.AutoUpdate.MachineImageVersion = true
+				Expect(testClient.Patch(ctx, shoot, patch)).To(Succeed())
 
-			Expect(kutil.SetAnnotationAndUpdate(ctx, testClient, shoot, v1beta1constants.GardenerOperation, v1beta1constants.ShootOperationMaintain)).To(Succeed())
+				Expect(kutil.SetAnnotationAndUpdate(ctx, testClient, shoot, v1beta1constants.GardenerOperation, v1beta1constants.ShootOperationMaintain)).To(Succeed())
+
+				Expect(waitForExpectedMachineImageMaintenance(ctx, logger, testClient, shoot, highestAMD64MachineImage, true, time.Now().Add(time.Second*20))).To(Succeed())
+			})
+
+			It("Shoot machine image must be updated in maintenance time: AutoUpdate.MachineImageVersion == false && expirationDate applies", func() {
+				// expire the Shoot's machine image
+				Expect(patchCloudProfileForMachineImageMaintenance(ctx, testClient, shoot.Spec.CloudProfileName, testMachineImage, &expirationDateInThePast, &deprecatedClassification)).To(Succeed())
 
-			Expect(waitForExpectedMachineImageMaintenance(ctx, logger, testClient, shoot, highestShootMachineImage, true, time.Now().Add(time.Second*20))).To(Succeed())
+				Expect(kutil.SetAnnotationAndUpdate(ctx, testClient, shoot, v1beta1constants.GardenerOperation, v1beta1constants.ShootOperationMaintain)).To(Succeed())
+
+				Expect(waitForExpectedMachineImageMaintenance(ctx, logger, testClient, shoot, highestAMD64MachineImage, true, time.Now().Add(time.Minute*1))).To(Succeed())
+			})
 		})
 
-		It("Shoot machine image must be updated in maintenance time: AutoUpdate.MachineImageVersion == false && expirationDate applies", func() {
-			// expire the Shoot's machine image
-			Expect(patchCloudProfileForMachineImageMaintenance(ctx, testClient, shoot.Spec.CloudProfileName, testMachineImage, &expirationDateInThePast, &deprecatedClassification)).To(Succeed())
+		Context("Worker pool machine's architecture is arm64", func() {
+			BeforeEach(func() {
+				patch := client.MergeFrom(shoot.DeepCopy())
+				shoot.Spec.Provider.Workers[0].Machine.Architecture = pointer.String("arm64")
+				Expect(testClient.Patch(ctx, shoot, patch)).To(Succeed())
```

Comment:
Instead of having two contexts with changes to the only existing worker pool, I think the test would be more valuable if you just added a second worker pool for ARM only (but with the same machine image name) and then check for the correctly updated version.